cloud: add migration docs for TiDB Cloud Essential (#21849)
pingcap/docs: tidb-cloud/migrate-from-mysql-using-data-migration.md

# Migrate MySQL-Compatible Databases to TiDB Cloud Using Data Migration

> This document guides you through migrating your MySQL databases from Amazon Aurora MySQL, Amazon RDS, Azure Database for MySQL - Flexible Server, Google Cloud SQL for MySQL, or self-managed MySQL instances to <CustomContent plan="dedicated">{{{ .dedicated }}}</CustomContent><CustomContent plan="essential">{{{ .essential }}}</CustomContent>using the Data Migration feature in the [TiDB Cloud console](https://tidbcloud.com/).

Reviewer: ```suggestion
This document guides you through migrating your MySQL databases from Amazon Aurora MySQL, Amazon RDS, Azure Database for MySQL - Flexible Server, Google Cloud SQL for MySQL, or self-managed MySQL instances to <CustomContent plan="dedicated">{{{ .dedicated }}}</CustomContent><CustomContent plan="essential">{{{ .essential }}}</CustomContent> using the Data Migration feature in the [TiDB Cloud console](https://tidbcloud.com/).
```

You can create up to 200 migration jobs on {{{ .dedicated }}} clusters for each organization. To create more migration jobs, you need to [file a support ticket](/tidb-cloud/tidb-cloud-support.md).

</CustomContent>
<CustomContent plan="essential">

You can create up to 100 migration jobs on {{{ .essential }}} clusters for each organization. To create more migration jobs, you need to [file a support ticket](/tidb-cloud/tidb-cloud-support.md).

</CustomContent>

### Filtered out and deleted databases

- The system databases will be filtered out and not migrated to TiDB Cloud even if you select all of the databases to migrate. That is, `mysql`, `information_schema`, `performance_schema`, and `sys` will not be migrated using this feature.

- When you delete a cluster in TiDB Cloud, all migration jobs in that cluster are automatically deleted and not recoverable.

<CustomContent plan="essential">

### Limitations of Alibaba Cloud RDS

When using Alibaba Cloud RDS as a data source, every table must have an explicit primary key. For tables without one, RDS appends a hidden primary key to the binlog, which leads to a schema mismatch with the source table and causes the migration to fail.

> ### Limitations of Alibaba Cloud PolarDB-X

Reviewer 2: If it is not fully verified, can we delete this section?

### Availability

- Currently, the Data Migration features is not available for {{{ .starter }}}.
<CustomContent plan="dedicated">
- If you don't see the [Data Migration](/tidb-cloud/migrate-from-mysql-using-data-migration.md#step-1-go-to-the-data-migration-page) entry for your {{{ .dedicated }}}cluster in the [TiDB Cloud console](https://tidbcloud.com/), the feature might not be available in your region. To request support for your region, contact [TiDB Cloud Support](/tidb-cloud/tidb-cloud-support.md).

> </CustomContent> 

Reviewer: ```suggestion

<CustomContent plan="dedicated">
- If you don't see the [Data Migration](/tidb-cloud/migrate-from-mysql-using-data-migration.md#step-1-go-to-the-data-migration-page) entry for your {{{ .dedicated }}} cluster in the [TiDB Cloud console](https://tidbcloud.com/), the feature might not be available in your region. To request support for your region, contact [TiDB Cloud Support](/tidb-cloud/tidb-cloud-support.md).
</CustomContent> 

```

### Availability

> - Currently, the Data Migration features is not available for {{{ .starter }}}.

Reviewer: ```suggestion
- Currently, the Data Migration feature is not available for {{{ .starter }}}.
```

<CustomContent plan="dedicated">

> - If you don't see the [Data Migration](/tidb-cloud/migrate-from-mysql-using-data-migration.md#step-1-go-to-the-data-migration-page) entry for your {{{ .dedicated }}} cluster in the [TiDB Cloud console](https://tidbcloud.com/), the feature might not be available in your region. To request support for your region, contact [TiDB Cloud Support](/tidb-cloud/tidb-cloud-support.md).

Reviewer: to fix the rendering issue of CustomContent

</details>

<details>
<summary> Configure Alibaba Cloud RDS MySQL</summary>

1. In the [ApsaraDB RDS console](https://rds.console.aliyun.com/), select the region of your instance, and then click the ID of your RDS for MySQL instance.

2. In the left navigation pane, click **Parameters**, search for each parameter, and then set the following values:

    - `binlog_row_image`: `FULL`

3. In the left navigation pane, click **Backup and Restoration**, and then select **Backup Strategy**. To ensure DM can access consecutive binlog files during migration, configure the backup strategy with the following constraints:

    - Retention Period: Set to at least 3 days (7 days recommended).

    - Retained Files: Ensure the "Max number of files" is sufficient to prevent older logs from being overwritten prematurely.

    - Storage Safeguard: Monitor Storage Usage closely. Note that RDS will automatically purge the earliest binlogs if the disk space usage reaches the system threshold, regardless of the retention period setting.

4. After applying the changes (and restarting if needed), connect to the instance and run the `SHOW VARIABLES` statement in this section to verify the configuration.

> For detailed instructions, see [Set instance parameters](https://www.alibabacloud.com/help/en/rds/apsaradb-rds-for-mysql/modify-the-parameters-of-an-apsaradb-rds-for-mysql-instance)

Reviewer: ```suggestion
For more information, see [Set instance parameters](https://www.alibabacloud.com/help/en/rds/apsaradb-rds-for-mysql/modify-the-parameters-of-an-apsaradb-rds-for-mysql-instance).
```

| `ALTER`  | Tables | Modifies table definitions when schema changes |
| `DROP`   | Databases, Tables | Removes objects during schema sync |
| `INDEX`  | Tables | Creates and modifies indexes |
| `CREATE VIEW`  | Views | Creates views used by migration |

> 

Reviewer: Please make the following updates in this doc

 From: ```click **Data** > **Migration** in the left navigation pane```

To: ```click **Data** > **Data Migration** in the left navigation pane```

    - **Data source**: the data source type.
    - **Connectivity method**: select a connection method for your data source based on your security requirements and cloud provider:

        <CustomContent plan="dedicated">

        - **Public IP**: available for all cloud providers (recommended for testing and proof-of-concept migrations).
        - **Private Link**: available for AWS and Azure only (recommended for production workloads requiring private connectivity).
        - **VPC Peering**: available for AWS and Google Cloud only (recommended for production workloads needing low-latency, intra-region connections with non-overlapping VPC/VNet CIDRs).

        </CustomContent>
        <CustomContent plan="essential">

>         - **Public IP**: available for all cloud providers (recommended for testing and proof-of-concept migrations).

Author: Public is correct.
